### PR TITLE
New host module and fixed memory collector for OS X 10.9 Mavericks

### PR DESCRIPTION
**Pull request by miniche**
Apple had integrated the WKdm algorithm to compress data in the RAM, in OS X 10.9 Mavericks.
Unused memory can be compressed to free more memory, without swapping.

I have tested this behavior on my MacBook Pro with 8 GB of RAM. Mac OS X can give 12 GB of memory to the system and to the applications, before starting swap on the hard drive/SSD! Just incredible.
During this process, the CPU occupation is higher.

You can read more about it on:
- http://www.apple.com/osx/advanced-technologies/
- http://terpconnect.umd.edu/~barua/matt-compress-tr.pdf

So, to do it, Apple had added new page types for the memory (see #85).
CentralReport can now analyse them and displays right data about memory.

I have added the "cr.host" module. It contains all code to determine the current host and to get information about it. It was previously in the "collector" module, which is destined to died in a next release.

### CODE REVIEW

**Comment by miniche** [comment]
Guys,

This pull request is an important fix to support the new OS X Mavericks. However, I had to change modules and functions which impact all OS.
In this case, we can't release a version "0.3.1", but a new "0.4.0", do you agree?

I will test this new code on Linux this afternoon, and I think it's better to merge this fix in master as soon as possible.

**Comment by miniche** [comment]
Team,

I've tested this improvement on Mac OS X 10.8 and 10.9, Debian 6 and 7, CentOS 6 and Ubuntu 13.04: works well.
So, what do you prefer for the version number?

And a review is welcome :wink:

File: centralreport/centralreport.py [hunk 1]
```diff
@@ -80,6 +81,9 @@ def run(self):
         CentralReport.starting_date = datetime.datetime.now()  # Starting date
         CentralReport.configuration = Config()  # Getting config object
 
+        # Getting data about the current host...
```
Comment (Ninir) [inline centralreport/centralreport.py]: Not useful comment detected! :smile:

**Comment by Ninir** [comment]
Seems good to me :)

**Comment by miniche** [comment]
Great :)

And about the version number, what do you prefer? (https://github.com/CentralReport/CentralReport/pull/87#issuecomment-27867470)

**Comment by Ninir** [comment]
It's a minor improvement, 0.4.0 is good for me :) ! :+1:

**Comment by miniche** [comment]
This pull request is not valid: this branch must be merged in develop, not in master. Replaced by #88.

**Pull request closed without merging**